### ISSUE TITLE
Ban certain roles in matchers

### ISSUE THREAD

**Issue by seanlafferty-ibm** [mskelton/eslint-plugin-playwright#420]
We have caught a lot of usage of this
```ts
await expect(page.getByRole("progressbar")).toBeHidden();
```
as a means of flow control in tests. It is often _very_ flaky because

1. Just because a progressbar is hidden _now_, does not mean it will be hidden 1 frame from now. Many are animated and do not render in the same exact frame as the network request fires
1. Sometimes a progressbar is _never_ rendered, if the request is fast enough
1. The amount of progressbars on screen is non-deterministic depending on what else is happening on the page (strict-mode violation)

We're often rewriting these progressbar assertions to instead assert what actually changed. Eg if we're testing a remote table sort, write an assertion like
```ts
await getByRole("button", {name: "sort"}).click()
await expect(
  // get rows
  // assert rows abide by the new sort
).toPass()
```

**Comment by mskelton** [comment]
I have some uses for this at work as well, so I like this idea a lot

**Comment by mskelton** [comment]
Will release on Sunday
